drivers/ethernet/eth_stm32_hal_priv.h

```diff
@@ -36,6 +37,11 @@ extern const struct device *eth_stm32_phy_dev;
 #elif defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32H7RSX)
 #define __eth_stm32_desc __attribute__((section(".eth_stm32_desc")))
 #define __eth_stm32_buf  __attribute__((section(".eth_stm32_buf")))
+#elif defined(CONFIG_SOC_SERIES_STM32MP13X)
+#define ETH_DMA_REGION  DT_INST_PHANDLE(0, memory_regions)
+#define ETH_DMA_LINKER_REGION_NAME LINKER_DT_NODE_REGION_NAME(ETH_DMA_REGION)
+#define __eth_stm32_desc ALIGN_32BYTES(__attribute__((__section__(ETH_DMA_LINKER_REGION_NAME))))
+#define __eth_stm32_buf  ALIGN_32BYTES(__attribute__((__section__(ETH_DMA_LINKER_REGION_NAME))))
```

Comment:
Use `Z_GENERIC_SECTION`.
```suggestion
#define ETH_DMA_REGION  DT_INST_PHANDLE(0, memory_regions)
#define ETH_SECTION Z_GENERIC_SECTION(LINKER_DT_NODE_REGION_NAME_TOKEN(ETH_DMA_REGION))
#define __eth_stm32_desc ALIGN_32BYTES(ETH_SECTION)
#define __eth_stm32_buf  ALIGN_32BYTES(ETH_SECTION)
```

(also, prefer `__aligned(32)` instead of the STM32-specific `ALIGN_32BYTES` if possible)